### ISSUE TITLE
Found usage of PostGetter which was earlier removed

### ISSUE THREAD

**Issue by szepeviktor** [timber/timber#2433]
https://github.com/timber/timber/blob/2.x/lib/Post.php#L379-L381

## Question

I do not know what to do instead in `check_post_id()`.

**Comment by acobster** [comment]
Thanks @szepeviktor, this method was supposed to be removed in THE PURGE (#2354) but I guess it managed to survive.

**Comment by szepeviktor** [comment]
Just run PHPStan and it help keeping Timber healthy.

**Comment by acobster** [comment]
@szepeviktor I imagine I would have if PHPStan were working for me right now... 😛 

```
$ vendor/bin/phpstan
Note: Using configuration file /srv/www/timber/phpstan.neon.
⚠️  You're using a deprecated config option autoload_files. ⚠️️

You might not need it anymore - try removing it from your
configuration file and run PHPStan again.

If the analysis fails, there are now two distinct options
to choose from to replace autoload_files:
1) scanFiles - PHPStan will scan those for classes and functions
   definitions. PHPStan will not execute those files.
2) bootstrapFiles - PHPStan will execute these files to prepare
   the PHP runtime environment for the analysis.

Read more about this in PHPStan's documentation:
https://phpstan.org/user-guide/discovering-symbols

PHP Fatal error:  Class 'WP_Upgrader_Skin' not found in /srv/www/timber/vendor/php-stubs/wp-cli-stubs/wp-cli-stubs.php on line 2925
Fatal error: Class 'WP_Upgrader_Skin' not found in /srv/www/timber/vendor/php-stubs/wp-cli-stubs/wp-cli-stubs.php on line 2925
```

This happens on `2.x` for me (inside VVV of course). Is this worthy of its own issue? Is it possible it's a bug in wp-cli-stubs?

**Comment by szepeviktor** [comment]
I think you are missing my last PR :)
Please get those new commit!

Then you'll get this PHPStan config: https://github.com/timber/timber/blob/2.x/phpstan.neon

**Comment by acobster** [comment]
My mistake, pulling `2.x` fixed it! Found 79 errors; I'll start working on those in a separate PR. Thanks!

**Comment by szepeviktor** [comment]
I gladly help.

Some interface thingies need PHP generics.

**Comment by acobster** [comment]
I hope you don't mean the [evil](https://github.com/ircmaxell/PhpGenerics) kind. 😈

**Comment by szepeviktor** [comment]
:)

> TL;DR: don't use this
